test/unittest/tests/block/test_block_session_size.cpp

```diff
@@ -0,0 +1,277 @@
+    while (curr != NULL) {
+        WT_SIZE *tmp = curr;
+        curr = curr->next[0];
+        __wt_free(nullptr, tmp);
+    }
+}
+void
+validate_size_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
+{
+    REQUIRE(bms != nullptr);
+    if (bms->sz_cache_cnt == 0)
+        REQUIRE(bms->sz_cache == nullptr);
+
+    REQUIRE(bms->sz_cache_cnt == expected_items);
+    WT_SIZE *curr = bms->sz_cache;
+    for (int i = 0; i < expected_items; i++) {
+        validate_size_block(curr);
+        curr = curr->next[0];
+    }
+    REQUIRE(curr == nullptr);
+}
+
+void
+validate_and_free_size_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
+{
+    validate_size_list(bms, expected_items);
+    free_size_list(bms);
+}
+
+TEST_CASE("Block session: __block_size_alloc", "[block_session_size]")
+{
+    WT_SIZE *sz = nullptr;
+
+    REQUIRE(__ut_block_size_alloc(nullptr, &sz) == 0);
+    validate_and_free_size_block(sz);
+}
+
+TEST_CASE("Block session: __block_size_prealloc", "[block_session_size]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+
+    SECTION("Allocate zero size blocks")
+    {
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 0) == 0);
+        validate_and_free_size_list(bms, 0);
+    }
+
+    SECTION("Allocate one size block")
+    {
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 1) == 0);
+        validate_and_free_size_list(bms, 1);
+    }
+
+    SECTION("Allocate multiple size blocks")
+    {
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 3) == 0);
+        validate_and_free_size_list(bms, 3);
+    }
+
+    SECTION("Allocate blocks on existing cache")
+    {
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 3) == 0);
+        validate_size_list(bms, 3);
+
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 0) == 0);
+        validate_size_list(bms, 3);
+
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 2) == 0);
+        validate_size_list(bms, 3);
+
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 5) == 0);
+        validate_and_free_size_list(bms, 5);
+    }
+}
+
+TEST_CASE("Block session: __wti_block_size_alloc", "[block_session_size]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+
+    SECTION("Allocate with null block manager session and no size cache")
+    {
+        std::shared_ptr<MockSession> session_no_bm = MockSession::buildTestMockSession();
+        WT_SIZE *sz;
+
+        REQUIRE(__wti_block_size_alloc(session_no_bm->getWtSessionImpl(), &sz) == 0);
+        validate_and_free_size_block(sz);
+
+        REQUIRE(__wti_block_size_alloc(session->getWtSessionImpl(), &sz) == 0);
+        validate_and_free_size_block(sz);
+    }
+
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
```

Comment:
`bms` is needed for the test below this line, but not the one above. However, when executing the first test SECTION (above) `bms` will be created after the test, and it is not needed anyway.

I suggest splitting this `TEST_CASE` into two: one containing the first `SECTION` from this `TEST_CASE`, and the second containing every other `SECTION` from this `TEST_CASE` to make things simpler and cleaner. Every `SECTION` in the (new) second `TEST_CASE` will then require `bms`.

Reply:
It will also then be easy to pull common setup code, eg
```
        WT_SIZE *sz;
        REQUIRE(__wti_block_size_alloc(session->getWtSessionImpl(), &sz) == 0);
```
out from each `SECTION` to the `TEST_CASE` level.

Reply:
I have done this now, please lmk if it looks good to you [6e7c584](https://github.com/wiredtiger/wiredtiger/pull/10982/commits/6e7c5842a48870d4129bbb55cf4c714db0ed2089)

Reply:
Thanks!